cmd/clusterawsadm/cmd/alpha/bootstrap/bootstrap.go

```diff
@@ -263,33 +264,25 @@ func generateAWSDefaultProfile() *cobra.Command {
 	return newCmd
 }
 
-func getCredentialsFromEnvironment() (*awsCredential, error) {
+func getCredentialsFromDefaultChain() (*awsCredential, error) {
 	creds := awsCredential{}
-
-	region, err := getEnv("AWS_REGION")
-	if err != nil {
-		return nil, err
-	}
-	creds.Region = region
-
-	accessKeyID, err := getEnv("AWS_ACCESS_KEY_ID")
+	conf := aws.NewConfig()
+	chain := defaults.CredChain(conf, defaults.Handlers())
```

Comment:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/05c39e93d8b035096e917df1e884b1c771f80969/vendor/github.com/aws/aws-sdk-go/aws/defaults/defaults.go#L88-L92 This comment says that this method shouldn't be used directly. 

Before merging this in, we should validate a few scenarios:
1. Does this work on an AWS instance and does it use an IAM role if available?
2. Is there a way to use [STS tokens](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/496) later on?
3. If a user has a single profile/credentials, does this effectively copy the local secrets?
4. Is it still possible to specify environment variable that override the local secret?

If all the above can be validated, the approach you proposed should probably be generalized also in the actuators.

Reply:
No, we don't need to do anything in the actuators.

Because we don't call explicit credentials, the SDK defaults to using the default credential chain anyway.

Reply:
Using the method directly is reasonable here, because the way we want to get the credentials into a secret is a bit hacky, and we just want to pretend that the injected secret came from the same source it would normally do if it wasn't in a pod/container.

Reply:
Right, that's the current behavior. We'd still need to revisit it or find other solutions for [credential management during pivot](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/297) and [after pivot](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/201). If there is a way to specify temporary bootstrap-only credentials, the approach used above can be used there too 😄.

Reply:
Let's not confuse these things, because we do not know the true final environments, and we should base this on how SDKs interact with IAM in practice, as well as upcoming SDK changes around OIDC.

Reply:
the tl;dr after chatting about this:

Need a doc/KEP on:
* the dev and management cluster use cases
* how clusterawsadm, beyond the cloudformation bootstrap, is for the dev / solution architect bootstrapping only from their computer
* the IAM credential provider chain, and how we expect it to be used in practice (e.g. instance metadata via kube2iam, OIDC etc...)